lib/std/array_list.zig

```diff
@@ -5,6 +5,10 @@ const testing = std.testing;
 const mem = std.mem;
 const Allocator = mem.Allocator;
 
+/// Safe, resizable array type.
```

Comment:
That's fair. I wanted to characterize it somewhat like std::vector, since that's what it seems to be. It's safer than just passing an unsized array around. Maybe just leave the "List of elements" part? I feel like this should be where an example for canonical use goes.